benchmark/v4.0-performance-benchmarking-with-tpch.md

```diff
@@ -0,0 +1,200 @@
+
+        ```bash
+        tiup bench tpch prepare \
+          --host ${tidb_v4_host} --port ${tidb_v4_port} --db tpch_10 --password ${password} \
+          --sf 10 \
+          --tiflash \
+          --analyze --tidb_build_stats_concurrency 8 --tidb_distsql_scan_concurrency 30
+        ```
+
+3. Execute the TPC-H query.
+
+    1. Download the TPC-H SQL query file:
+
+        {{< copyable "" >}}
+
+        ```bash
+        git clone https://github.com/pingcap/tidb-bench.git && cd tpch/queries
+        ```
+
+    2. Execute the query and record the executing time of the query.
+
+        * For TiDB v3.0, use the MySQL client to connect to TiDB, execute the query, and record the v3.0 query processing time.
+        * For TiDB v4.0, use the MySQL client to connect to TiDB, and choose one of the following operations based on where data is read from:
+            * If data is read only from TiKV, set `set @@session.tidb_isolation_read_engines = 'tikv,tidb';`, execute the query, and record the v4.0 query processing time.
+            * If data is read from TiKV and TiFlash automatically based on intelligent choice, set `set @@session.tidb_isolation_read_engines = 'tikv,tiflash,tidb';`, execute the query, and record the v4.0 query processing time.
+
+4. Extract and organize the data of query processing time.
+
+### Test result
+
+| Query ID |  v3.0  |  v4.0 TiKV Only |  v4.0 TiKV/TiFlash Automatically |
+| :-------- | :----------- | :------------ | :-------------- |
+| 1       |    7.78s   |      7.45s  |      2.09s    |
+| 2       |    3.15s   |      1.71s  |      1.71s    |
+| 3       |    6.61s   |      4.10s  |      4.05s    |
+| 4       |    2.98s   |      2.56s  |      1.87s    |
+| 5       |   20.35s   |      5.71s  |      8.53s    |
+| 6       |    4.75s   |      2.44s  |      0.39s    |
+| 7       |    7.97s   |      3.72s  |      3.59s    |
+| 8       |    5.89s   |      3.22s  |      8.59s    |
+| 9       |   34.08s   |     11.87s  |     15.41s    |
+| 10      |    4.83s   |      2.75s  |      3.35s    |
+| 11      |    3.98s   |      1.60s  |      1.59s    |
+| 12      |    5.63s   |      3.40s  |      1.03s    |
+| 13      |    5.41s   |      4.56s  |      4.02s    |
+| 14      |    5.19s   |      3.10s  |      0.78s    |
+| 15      |   10.25s   |      1.82s  |      1.26s    |
+| 16      |    2.46s   |      1.51s  |      1.58s    |
+| 17      |   23.76s   |     12.38s  |      8.52s    |
+| 18      |   17.14s   |     16.38s  |     16.06s    |
+| 19      |    5.70s   |      4.59s  |      3.20s    |
+| 20      |    4.98s   |      1.89s  |      1.29s    |
+| 21      |   11.12s   |      6.23s  |      6.26s    |
+| 22      |    4.49s   |      3.05s  |      2.31s    |
+
+![TPC-H](/media/tpch_v4vsv3.png)
+
+In the performance diagram above:
+
++ Blue lines represent v3.0;
++ Red lines represent v4.0 (data read only from TiKV);
++ Yellow lines represent v4.0 (data read from TiKV and TiFlash automatically based on intelligent choice).
```

Comment:
We use the term "smart selection" instead of "intelligent choice" in this doc:
https://pingcap.com/docs/stable/tiflash/use-tiflash/#smart-selection

Reply:
This  TiFlash overview document uses `Intelligent choice` https://pingcap.com/docs/stable/tiflash/tiflash-overview/#intelligent-choice